### ISSUE TITLE
feat(backend): Error handling framework

### ISSUE THREAD

**Issue by KavikaPalletenne** [devsoc-unsw/chaos#473]

**Comment by KavikaPalletenne** [comment]
Some examples:
https://github.com/tokio-rs/axum/blob/main/examples/error-handling/src/main.rs

**Comment by fritzrehde** [comment]
What exactly do you want? Is this about logging or reporting terminal errors?

**Comment by KavikaPalletenne** [comment]
[`anyhow`](https://crates.io/crates/anyhow) does not seem to be the best crate for our use case. It hides every type of error inside it's own `anyhow::Error` type. However, we need to return different server responses based on each error, and thus, the [`thiserror`](https://crates.io/crates/thiserror) crate seems like a better candidate.

**Comment by fritzrehde** [comment]
Yup, therror is great! It's usually used more in libraries, where you want to expose your own error types to users of your library so those users can figure out which of your errors to handle in what way. But I'd argue anyhow is the "I don't want to think too much about errors and just want to print sth to the screen"-type of solution, so organizing different failure cases in thiserror is nicer in my opinion for a sizeable project like Chaos.
